prime-router/src/main/kotlin/history/db/DeliveryDatabaseAccess.kt

```diff
@@ -172,7 +172,8 @@ class DeliveryDatabaseAccess(val db: DatabaseAccess = BaseEngine.databaseAccessS
         if (sentReportIdsForReceiver.isEmpty()) {
```

Comment:
With this change it seems the only uses for the output of `fetchReportIdsForReceiverAndTask` is to check if the resulting dataset is empty. Could be a spot for further optimization?

Reply:
I think you're correct, but I left this is so that I could measure this in PROD.  It might be that finding that it's empty earlier might speed things up.